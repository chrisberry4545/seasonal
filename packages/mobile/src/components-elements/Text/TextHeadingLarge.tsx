import React, { FC } from 'react';
import { Text, TextStyle } from 'react-native';
import { headingBase } from './SharedTextStyles';

import {
  IText
} from './Text.interface';

export const styleTextHeadingLarge: TextStyle = {
  ...headingBase,
  fontSize: 30,
  lineHeight: 40
};

export const TextHeadingLarge: FC<IText> = ({
  children,
  style
}) => (
  <Text style={ [styleTextHeadingLarge, style] }>
    { children }
  </Text>
);
