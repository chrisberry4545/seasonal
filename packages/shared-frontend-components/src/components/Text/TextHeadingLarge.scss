.c-text-heading-large {
  font-size: 2rem;
  font-weight: 300;
}
