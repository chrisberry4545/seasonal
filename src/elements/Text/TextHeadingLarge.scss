.c-text-heading-large {
  font-size: 2rem;
  font-weight: 300;
  margin: 0;
}
