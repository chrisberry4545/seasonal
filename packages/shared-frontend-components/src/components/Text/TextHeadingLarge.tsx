import React, { FC } from 'react';
import './TextHeadingLarge.scss';
import { IText } from './Text.interface';

export const TextHeadingLarge: FC<IText> = ({
  className,
  children
}) => (
  <span className={`${(className || '')} c-text-heading-large`}>
    { children }
  </span>
);
