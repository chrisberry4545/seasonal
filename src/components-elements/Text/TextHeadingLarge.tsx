import React, { FC } from 'react';
import './TextHeadingLarge.scss';
import { ITextInterface } from './Text.interface';

export const TextHeadingLarge: FC<ITextInterface> = ({
  className,
  children
}) => (
  <span className={`${(className || '')} c-text-heading-large`}>
    { children }
  </span>
);
