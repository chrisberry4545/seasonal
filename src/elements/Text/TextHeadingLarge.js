import React from 'react';
import './TextHeadingLarge.scss';

export const TextHeadingLarge = ({ className, children }) => (
  <span className={`${(className || '')} c-text-heading-large`}>
    { children }
  </span>
);
