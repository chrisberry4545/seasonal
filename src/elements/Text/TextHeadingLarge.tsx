import React, { SFC } from 'react';
import './TextHeadingLarge.scss';
import { TextInterface } from './Text.interface';

export const TextHeadingLarge: SFC<TextInterface> = ({
  className,
  children
}) => (
  <span className={`${(className || '')} c-text-heading-large`}>
    { children }
  </span>
);
