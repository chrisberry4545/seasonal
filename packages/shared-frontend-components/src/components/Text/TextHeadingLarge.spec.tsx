import React from 'react';
import { TextHeadingLarge } from './TextHeadingLarge';
import { shallow, ShallowWrapper } from 'enzyme';

describe('<TextHeadingLarge />', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(
      <TextHeadingLarge
        className='class'>
          Text
      </TextHeadingLarge>
    );
  });

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());
});
