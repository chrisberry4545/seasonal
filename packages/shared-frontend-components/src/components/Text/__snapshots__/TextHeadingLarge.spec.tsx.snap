// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextHeadingLarge /> renders correctly 1`] = `
<span
  className="class c-text-heading-large"
>
  Text
</span>
`;
